src/Ng/CommandHelpers.cs

```diff
@@ -140,11 +236,7 @@ public static bool GetVerbose(IDictionary<string, string> arguments)
         public static bool GetBool(IDictionary<string, string> arguments, string argumentName, bool defaultValue)
         {
             string argumentValue;
-            if (!arguments.TryGetValue("-" + argumentName, out argumentValue))
-            {
-                return defaultValue;
-            }
-
+            TryGetArgument(arguments, argumentName, out argumentValue);
             return string.IsNullOrEmpty(argumentValue) ? defaultValue : argumentValue.Equals("true", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
It seems like there are 2 cases here:
1. Parameter is not specified. In this case, we should fall back to the default value.
2. Parameter is specified, but specified value is not a valid value. In this case, we should fail.

Is this the behaviour that is expected?

---
In reply to: [71022687](https://github.com/NuGet/NuGet.Services.Metadata/pull/77#discussion_r71022687) [](ancestors = 71022687)